CMakeLists.txt

```diff
@@ -202,6 +202,17 @@ install (FILES src/cmake/modules/FindOpenImageIO.cmake
          DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/cmake/Modules
          COMPONENT cmake)
 
+# Export the configuration files. There are also library-specific config
+# exports in the CMakeLists.txt of libOpenImageIO.
+include (CMakePackageConfigHelpers)
+write_basic_package_version_file (
+    "${PROJECT_NAME}ConfigVersion.cmake"
+    VERSION ${PROJECT_VERSION}
```

Comment:
Yes, would you mind submitting a separate PR from this one, that proposes what you think are the best practices? 

I appreciate the help. I've only recently modernized my own CMake knowledge which had been somewhat stagnant since OIIO's build system was written in the CMake 2.8 era, and now feel fairly fluent in ~3.12 and updated OIIO's build scripts as I went. But to be honest, the new config techniques are one of the areas I'm still not confident with.

Reply:
Sure thing, I'll try to do it asap.

I didn't see what @kdt3rd did for openEXR but it is really good news he added that!!

The only "fancy" thing that I have here (besides using variables for better maintainability) is the template file `Config.in.cmake` that includes the separate file with the targets.
This is even recommended in the [cmake official tutorial](https://cmake.org/cmake/help/v3.0/manual/cmake-packages.7.html#creating-packages), because you may need to add some other stuff in the config. For example, if we ever switch to use `openEXR::openEXR` instead of using the `${OPENEXR_INCLUDE_DIRS}` etc, then inside the config we need that line to find the dependency (as I explained in the previous message).

Reply:
Ah, yes, makes sense. OpenEXR has almost no dependencies other than zlib which is built into cmake, so this makes sense that it didn't need a custom one, but can just use the one for the semi-internal targets of IlmBase. And as soon as 2.4 of exr is deployed more widely, oiio can switch to just use the OpenEXR::OpenEXR target and config mechanism, although I think that may be a little while before the various distributions update, so wouldn't recommend that just yet.

Reply:
Yes, OpenImageIO_Util is a separate library, we definitely would want a separate imported target for it. You never need both -- libOpenImageIO contains everything that's in libOpenImageIO_Util just because there's generally useful things there. But sometimes people just need the util part and don't need the actual image handling stuff (and its many dependencies), so as a convenience we made this second library that's just a subset.